### DISCUSSION TITLE
trailingstoploss_tp.ini with 0 second profit stop loss timeout creating Market instead of Limit orders?

### DISCUSSION THREAD

**Discussion by MisterMelons** [cyberjunky/3commas-cyber-bots#208] [General]
Hi,

Based off the description for **Profit** in the documentation for **trailingstoploss_tp.py**, I assumed that a 0 second sl-timeout would be creating Limit orders:

_"Advantage is the limit order is already in the orderbook, so on a sudden pump the deal will close as with the 3C TTP functionality the market order could be left unfilled."_

I just noticed today that all of my 0 second sl-timeout orders have been closing as **Market** orders and not Limit orders. I found out  because a deal closed below my first and only stop loss for a deal. Has something changed since the documentation was created, am I reading it wrong or should I be using a setting of `"sl-timeout": = ""` instead of `"sl-timeout": = "0"`?

Thanks!

**Comment by MisterMelons** [comment]
Well, `"sl-timeout": = ""` doesn't work. It breaks the code. I'll just wait for a reply...

**Comment by MisterMelons** [comment]
I wanted to clarify that I'm not just looking at the sell order details in 3Commas to see that it was a Market order. I realize that 3Commas may label deals closed by a "Stop Loss" as "Market" even if they are Limit orders.

I was bringing this up because my deal closed at a price that was lower than the Stop Loss price that was set by the script. Screenshots and logs can be provided if needed.

Thanks!

**Comment by MisterMelons** [comment]
Seriously... is there a way to create a Limit order for the first Stop Loss? I'm trying to lock in a minimum profit before allowing the deal to run, but I'm finding no success. I don't understand why a 0 second Stop Loss creates a Market order...

**Comment by amargedon** [comment]
You are trying to apply the logic of an order for the profit side, on the logic for an order on the safety side and it doesn't work like that. A TP order can be moved higher, resulting in a limit order in the orderbook.

A stoploss order can only be placed under conditions and not as market order according to the existing functionality of 3C. The sl-timeout option is used by 3C for the stoploss. And because of the market order, it can (and will sometimes) sell it lower than the configured stoploss. This script does not replace the stoploss logic of 3C, it improves the TP and Safety Order logic only.

**Comment by MisterMelons** [comment]
Understood. Thank you.
